File: backend/src/main/java/com/example/demo/models/AuditLog.java
package com.example.demo.models;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Table(name = "audit_log")
@Data
@NoArgsConstructor
public class AuditLog {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "log_id")
    private Long logId;

    @Column(name = "event_time", nullable = false)
    private LocalDateTime eventTime = LocalDateTime.now();

    @Column(name = "level", nullable = false)
    private String level = "INFO";

    @Column(name = "event_type", nullable = false, length = 50)
    private String eventType;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private AppUser user;

    @Column(name = "module", length = 100)
    private String module;

    @Column(name = "result", length = 20)
    private String result;

    @Column(name = "message")
    private String message;
}

File: frontend/lib/models/audit_log.dart
class AuditLog {
  final int? id;
  final String? eventTime;
  final String? level;
  final String? eventType;
  final int? userId;
  final String? module;
  final String? result;
  final String? message;

  AuditLog({
    this.id,
    this.eventTime,
    this.level,
    this.eventType,
    this.userId,
    this.module,
    this.result,
    this.message,
  });

  factory AuditLog.fromJson(Map<String, dynamic> json) {
    return AuditLog(
      id: json['logId'],
      eventTime: json['eventTime'],
      level: json['level'],
      eventType: json['eventType'],
      userId: json['userId'],
      module: json['module'],
      result: json['result'],
      message: json['message'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      if (id != null) 'logId': id,
      'eventTime': eventTime,
      'level': level,
      'eventType': eventType,
      'userId': userId,
      'module': module,
      'result': result,
      'message': message,
    };
  }
}
